docs/specifications/tests/Syntax-TP/syntax06.md

```diff
@@ -1,54 +1,113 @@
-field should follow the rules of an e-mail address also defined in section
-3.4.1 of [RFC 2822](https://tools.ietf.org/html/rfc2822#section-3.4).
+in section 3.3.13 of [RFC 1035] and section 2.2 of [RFC 1912]. The RNAME
+field should follow the rules of an e-mail address also defined in 
+[RFC 5322, section 3.4.1].
 
-### Inputs
+## Inputs
 
-The domain name to be tested.
+* "Child Zone" - The domain name to be tested.
 
-### Ordered description of steps to be taken to execute the test case
+## Ordered description of steps to be taken to execute the test case
 
-1. Obtain a set of name server IP addresses using [Method4] and [Method5].
-2. Create a SOA query for the zone.
-3. Send the SOA query over UDP to each name server IP address until a
-   response is received or until the set is exhausted.
-4. Get the RNAME from the SOA record.
-5. Convert the first non-escaped "." to an "@" in the RNAME.
-6. De-escape the RNAME, converting a "\." to ".".
-7. Verify the result from step 3 and 4 according to the Addr-spec
-   specification in RFC 2822 section 3.4.1. If this verification fails,
-   this test case failes.
+1. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
 
+2. Create a SOA query for the apex of the *Child Zone* with "rd" flag unset.
 
-### Outcome(s)
+3. For each name server in *Name Server IP* do:
+   1. Send the SOA query over UDP to the name server.
+   2. If the name server does not respond, emit *[NO_RESPONSE]* and go
+      to next server.
+   3. If the name server responds with DNS response but does not 
+      include a SOA record in the answer section, emit 
+      *[NO_RESPONSE_SOA_QUERY]* and go to next server.
+   4. Extract the RNAME from the SOA record.
+   5. Convert the first "." without backslash quoting to an "@" in 
+      the RNAME.
+   6. Convert any backslash quoted "." to a single "." without quoting
+      (see section 5.1, 5.3 and 8 in [RFC1035] for the use of backslash).
+   7. If the converted string (mail address) does not meet the 
+      specification in [RFC 5322, section 3.4.1], then emit 
+      *[RNAME_RFC822_INVALID]* and go to next server.
+   8. Extract the domain part (to the right of "@") from the mail 
+      address.
+   9. Create an MX query for the domain part and send it to a resolving
+      name server. If the lookup does not return a DNS response with
+      with RCODE "NOERROR" emit *[RNAME_MAIL_DOMAIN_INVALID]*.
+   10. If the lookup in previous step returned MX record or records for 
+       the mail domain, then replace the mail domain with the mail 
+       exchange or exchanges from RDATA of the MX record or records in
+       next step.
+   11. Create address querie (A and AAAA) for the mail domain or mail
+       exchage in previous steps (multiple queries if there are multiple
+       mail exchanges). At least one query must successfully return
+       an IP address or else emit *[RNAME_MAIL_DOMAIN_INVALID]*.
```

Comment:
Skulle det vara rimligt att skriva ut vad som krävs av svarspaketen m.a.p. rcode, flaggor och sektioner, eller är det överflödigt?

Reply:
In this case we are sending the queries to a resolving name server and accepts what we get without checking flags.